src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1810,6 +1810,77 @@ internal MethodSymbol EnsureThrowIfNullFunctionExists(SyntaxNode syntaxNode, Syn
             Debug.Assert((object?)privateImplClass.GetMethod(PrivateImplementationDetails.SynthesizedThrowFunctionName)!.GetInternalSymbol() == internalSymbol.ThrowMethod);
             return internalSymbol;
         }
+
+        /// <summary>
+        /// Creates the ThrowSwitchExpressionException helper if needed.
+        /// </summary>
+        internal MethodSymbol EnsureThrowSwitchExpressionExceptionExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, DiagnosticBag? diagnostics)
```

Comment:
Can diagnostics ever be null? Same question for other helpers below

Reply:
@jcouv I had copied the existing signature for `ThrowIfNull`

https://github.com/dotnet/roslyn/blob/534c0e8d04e58044c66ef4f66eeafd255a18749d/src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs#L1790

But I agree we should change it to non-nullable.